### DISCUSSION TITLE
try to add custom javascript but failed, please help me to solve this situation

### DISCUSSION THREAD

**Discussion by LingzheM** [hexojs/hexo#5237] [Question / Answer]
my hexo version is `hexo-cli: 4.3.1` , node version is `node: 19.7.0`,  on mac `os: darwin 21.6.0 12.6`.
first, i add a directory called `javascript` under hexo source directory, and i changed the `skip-render` in hexo config file as
```
skip_render:
  - "javascript/**"
```
and i clone the `2048`(https://github.com/gabrielecirulli/2048) code into the javascript directory,
and i add a iframe tag into a .md file , the content is
```
<iframe src="/javascript/2048/index.html" width="100%" height="1000" style="border:1px solid black;">
</iframe>
```.
the situation is when i executed `hexo clean and hexo generate` and `hexo server`
when i open the page , it display that `Cannot GET /javascript/2048/index.html`, 
i want to how to add custom js file or css file in hexo, please help me to fix this question.

**Comment by stevenjoezhang** [comment]
Please provide your blog source code

**Reply by LingzheM** [reply]
thanks. my blog source code is 
```
---
title: 230706
abbrlink: 7971245b
date: 2023-07-03 18:45:47
tags: N
categories: N
---

<!-- more -->

<iframe src="/javascript/2048/index.html" width="100%" height="1000" style="border:1px solid black;">
</iframe>
```
the blog file's absolute path is `/new-blog/source/_post/230706.md`, the 2048  directory path is `/new-blog/source/javascript/2048`

**Reply by stevenjoezhang** [reply]
Run `hexo g` and see if `/javascript/2048/index.html` exist

**Reply by LingzheM** [reply]
after run `hexo g`, the output has `INFO  Generated: javascript/2048/index.html`

**Reply by stevenjoezhang** [reply]
Can you find it in `public` folder? Can you run another web server e.g. Nginx to see if it can be served?

**Reply by LingzheM** [reply]
ye, when into the public, there truly has `index.html`, and i use `readlink` to print the path is`/new-blog/public/javascript/2048/index.html`(i delete the begin part of users). sorry for i do not get point, is that i start another server and see 2048 can be insert into a html file?

**Reply by stevenjoezhang** [reply]
Yes, you can use Apache httpd or Nginx to serve your public folder, to see if it works.
